### DISCUSSION TITLE
Support for Typst compiler

### DISCUSSION THREAD

**Discussion by Sirra630** [quarto-dev/quarto-cli#5024] [Feature Requests]
Hey there,

I just wanted to give a shoutout to your team for creating such an awesome software! It's seriously changed the game for me  and I am eager to encourage my coworkers to adopt it as well. However,  some of my coworkers who are used to using Word are finding the math typing system (based on Latex) a bit too complicated.

I recently stumbled upon an alternative to Latex called  [Typst](https://github.com/typst/typst). It's an open-source document formatting system that offers a more intuitive way to write math and has some interesting design decisions. What I particularly like about Typst is the fast compilation time, which provides almost instant previews. This is possible because the team behind Typst has opted for an incremental compilation process and a compiler written in Rust.

Although the project is still young and the design will likely evolve, a simple writer/reader has already been implemented in Pandoc (see [here](https://github.com/jgm/pandoc/issues/8713#issue-1634871522)). I was wondering if it would be possible to have support for Typst in Quarto in the future, maybe as a dedicated PDF engine?

Thanks for considering this idea!

**Comment by jjallaire** [comment]
Hi @Sirra630, thanks for the feedback! We are working on `typst` support in Quarto at this very moment! This will show up in the pre-release v1.4 builds once v1.3 has shipped (likely by end of April). Will post back here when there is something for you to try.

**Reply by Sirra630** [reply]
That's amazing, thank you for this great work !

**Comment by baptiste** [comment]
I was thinking about possible integrations, should Typst become mature enough that it's a robust alternative to TeX (still a long way to go). I could imagine 3 different routes for quarto-typst interactions:

1. Author in Typst (optionally, of course),  which is pre-processed into a standardised qmd intermediate input to be treated as usual by Quarto. \
   This has the advantage of providing some access to novel features, such as the programmable layout and drawing features of typst's mini-language (which, incidentally, I wish had been picked from a standard one like Lua/Ocaml/etc. :S)

1. Author in qmd, process via pandoc using Typst instead of TeX as intermediate\
   Definitely the pragmatic way to go short-term, especially with pandoc's reader/writer support, but it doesn't make full use of Typst's potential, and the qmd/md-pandoc syntax may someday become a real bottleneck (cf. [Butterick's gripes with markdown](https://docs.racket-lang.org/pollen/second-tutorial.html#%28part._the-case-against-markdown%29))

1. Progressively evolve the qmd format to be more like Typst (probably difficult/impossible given the constraints of jupyter notebooks). Still, having such a mini-language built into the format could open new possibilities. Djot is another source of inspiration, to take the good things about markdown, but also go further, and I believe Typst and Pollen are two examples of an even more flexible paradigm (where the whole documents are programmable).

Anyway, just some thoughts about these exciting new developments!

**Comment by jjallaire** [comment]
Current work in progress on this is here: https://github.com/quarto-dev/quarto-cli/tree/feature/typst (you can try it out by installing the development version as noted in the PR description).

@baptiste Note that you can use raw blocks to include `typst` markup directly in a `.qmd`.

**Comment by mcanouil** [comment]
Closing this as part of Quarto 1.4 (currently in pre-release state), see <https://quarto.org/docs/prerelease/1.4/typst.html>.
